plugins/indexing/batching/module.go

```diff
@@ -0,0 +1,59 @@
+package batching
+
+import (
+	"bytes"
+
+	// "cosmossdk.io/collections"
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/sedaprotocol/seda-chain/plugins/indexing/log"
+	"github.com/sedaprotocol/seda-chain/plugins/indexing/types"
+	batchingtypes "github.com/sedaprotocol/seda-chain/x/batching/types"
+)
+
+const StoreKey = batchingtypes.StoreKey
+
+func ExtractUpdate(ctx *types.BlockContext, cdc codec.Codec, logger *log.Logger, change *storetypes.StoreKVPair) (*types.Message, error) {
+	if _, found := bytes.CutPrefix(change.Key, batchingtypes.DataResultsPrefix); found {
+		if change.Delete {
+			logger.Trace("skipping delete", "change", change)
+			return nil, nil
+		}
+
+		val, err := codec.CollValue[batchingtypes.DataResult](cdc).Decode(change.Value)
+		if err != nil {
+			return nil, err
+		}
+
+		data := struct {
+			ID             string `json:"result_id"`
+			DrID           string `json:"dr_id"`
+			Version        string `json:"version"`
+			BlockHeight    uint64 `json:"block_height"`
+			ExitCode       uint32 `json:"exit_code"`
+			GasUsed        uint64 `json:"gas_used"`
+			Result         []byte `json:"result"`
+			PaybackAddress string `json:"payback_address"`
+			SedaPayload    string `json:"seda_payload"`
+			Consensus      bool   `json:"consensus"`
```

Comment:
Would it make more sense to have this as a proper structure if we need to use it outside this instance? Or is it only used in this instance?

Reply:
Oh true. Why don't we just use `DataResult` type defined in `x/batching/types/batching.pb.go`?

Reply:
Those have the omitempty attribute and I’d prefer not to have to deal with that at the receiving end 😅

Reply:
Ah yes, you put it in the prompt 🙃